### ISSUE TITLE
ContentChild query does not work with string queries

### ISSUE THREAD

**Issue by bennylut** [angular/angular#6182]
When using the ContentChild annotation with a type query it works very well but when I try to use string query - for some reason the annotated field is always null. 

the following is a SSCCE that reproduce the problem:

```
import {Component, ContentChild} from 'angular2/core';
import {bootstrap} from 'angular2/platform/browser';

@Component({
  template: `
    <ng-content select="inner"></ng-content>
  `,
  selector: 'ctr'
})
class TestContainerComponent {
  @ContentChild('inner') inner;

  ngAfterContentInit() {
    console.log('inner: ', this.inner);
  }
}


@Component({
  template: `
    <ctr> <inner> SOME TEXT </inner> </ctr>
  `,
  selector: 'app',
  directives: [TestContainerComponent]
})
class TestAppComponent { }

function main() {
    return bootstrap(TestAppComponent).catch(err => console.error(err));
}

document.addEventListener('DOMContentLoaded', main);
```

When running the above code the result is: 
console: inner: null
view: SOME TEXT

although the API docs show a usage example with a type query, the type definition indicates that the query can be either a Type or a string (Type | string)

**Comment by ericmartinezr** [comment]
Have you tried with `<ng-content select="inner" #inner></ng-content>`?

**Comment by bennylut** [comment]
Yes, both this way:

```
@Component({
  template: `
    <ng-content #inner select="inner"></ng-content>
  `,
  selector: 'ctr'
})
export class TestContainerComponent {
  @ViewChild('inner') inner;

  ngAfterViewInit() {
    console.log('inner: ', this.inner);
  }
}
```
and this way: 

```
@Component({
  template: `
    <ng-content #inner select="inner"></ng-content>
  `,
  selector: 'ctr'
})
class TestContainerComponent {
  @ContentChild('inner') inner;

  ngAfterContentInit() {
    console.log('inner: ', this.inner);
  }
}
```

returns the same output..

**Comment by ericmartinezr** [comment]
@bennylut we are wrong! :P 

First of all the string selector must be a variable name (can't say for sure that it doesn't work with css selectors) and it must be in the content that will be passed through ng-content, not in ng-content itself.

Check this code example ([plnkr](http://plnkr.co/edit/etOkrmwT5w0egVEMizZM?p=preview))

```
@Component({
  selector : 'my-cmp',
  template : ''
})
class MyCmp {
  @ContentChild('someDiv') someDiv;
  constructor() {}
  
  ngAfterContentInit() {
    console.log(this.someDiv);
  }
}

@Component({
  selector: 'my-app',
  template : `
    <my-cmp>
      <div #someDiv></div>
    </my-cmp>
  `,
  directives : [MyCmp]
})
export class App {}

**Comment by bennylut** [comment]
I see, this is a conflicting behavior between ng-content's select and ContentChild (which I thought that are processed the same way..)

So there is no way to get the content passed to the container-component if this content has no defined type (i.e., not a component itself)?

**Comment by zoechi** [comment]
@bennylut what does `@ChildView` actually different than `@ContentView`? 
You could just inject `ElementRef` and access the children using `elementRef.nativeElement.innerHTML`. Your question doesn't explain what's you actually try to accomplish.

**Comment by evanplaice** [comment]
@zoechi According to the docs `ElementRef` should be used as a *last resort* because it touches the DOM directly and requires direct coupling to the rendering layer.

*Source: https://angular.io/docs/js/latest/api/core/ElementRef-class.html*

@bennylut Is right in that there's no 'safe' way to access and transform a components contents without adding additional (read unnecessary) complexity.

Is there some technical reason why a facade isn't provided that allows access the component's `@Content`?

**Comment by zoechi** [comment]
@evanplaice I'm aware of this but if he wants access to the DOM to manipulate it then he does direct DOM access anyway and then I guess he can just use `ElementRef` as well. I think this is not clear from the original question.

**Comment by evanplaice** [comment]
@zoechi Seems pretty clear to me.

Whether it takes place on a `@Component` or child `@Component`, @bennylut is attempting to select the and log the inner text.

```javascript
...
  template: `
    <ng-content select="inner"></ng-content>
  `,
...
  ngAfterContentInit() {
    console.log('inner: ', this.inner);
  }
...
```

`<ng-content>` passes the contents of the component through to the template. AFAIK, Angular2 provides no facility to access those contents.

**Comment by zoechi** [comment]
@evanplaice I really doubt @bennylut actually wants to log the HTML XD

**Comment by alxhub** [comment]
Hey @tbosch,

Ran into this too - here's a plunk reproducing the issue against RC: http://plnkr.co/edit/5BYYJx1hnTKifTEvFJXv?p=preview

**Comment by zoechi** [comment]
@alxhub It's working fine. The string parameter is not for CSS selectors but for one or more template variable names. http://plnkr.co/edit/YFY3MovqDXFy9cIaFTIv?p=preview

I think this issue can be closed. All is working as designed.

**Comment by pkozlowski-opensource** [comment]
Yeh, this issue starts to be "catch all" for all issues that sound _similar_ to sth around query. I believe that most of what is discussed here will be solved by #8563 as it will allow querying arbitrary child elements. Closing as a duplicate but if you believe that #8563 isn't covering your use-case please open a separate issue with clear description of what is missing (or comment in #8563). Thnx!
